dev/ProgressRing/ProgressRing.cpp

```diff
@@ -41,18 +47,98 @@ void ProgressRing::OnRangeBasePropertyChanged(const winrt::DependencyObject&, co
     UpdateSegment();
 }
 
+void ProgressRing::OnForegroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
+{
+    ApplyLottieAnimation();
+}
+
+void ProgressRing::OnBackgroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
+{
+    ApplyLottieAnimation();
+}
+
 void ProgressRing::OnStrokeThicknessPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
 {
+    ApplyLottieAnimation();
     UpdateRing();
 }
 
-winrt::Size ProgressRing::ComputeEllipseSize(double thickness, double actualWidth, double actualHeight)
+void ProgressRing::OnIsIndeterminatePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateStates();
+}
+
+void ProgressRing::ApplyLottieAnimation()
+{
+    if (auto&& player = m_player.get())
+    {
+        // ProgressRing only accounts for ActualWidth to ensure that it is always a circle
+        const float diameter = static_cast<float>(ActualWidth());
+        const auto size = winrt::Size({ diameter, diameter });
+
+        const auto [foregroundColor, backgroundColor] = [
+            foreground = Foreground().try_as<winrt::SolidColorBrush>(),
+            background = Background().try_as<winrt::SolidColorBrush>()]()
+        {
+            winrt::Color foregroundColor;
+            winrt::Color backgroundColor;
+
+            if (foreground)
+            {
+                foregroundColor = foreground.Color();
+            }
+            else
+            {
+                foregroundColor = SharedHelpers::FindResource(s_DefaultForegroundThemeResourceName, winrt::Application::Current().Resources()).as<winrt::SolidColorBrush>().Color();
+            }
+
+            if (background)
+            {
+                backgroundColor = background.Color();
+            }
+            else
+            {
+                backgroundColor = SharedHelpers::FindResource(s_DefaultBackgroundThemeResourceName, winrt::Application::Current().Resources()).as<winrt::SolidColorBrush>().Color();
+            }
+
+            return std::make_tuple(foregroundColor, backgroundColor);
+        }();
+       
+        
+        player.Source(winrt::make<ProgressRingLoading>(StrokeThickness(), size, foregroundColor, backgroundColor));
+    }
+}
+
+void ProgressRing::UpdateStates()
+{
+    if (IsIndeterminate())
+    {
+        winrt::VisualStateManager::GoToState(*this, s_IndeterminateStateName, true);
+
+        if (auto&& player = m_player.get())
+        {
+            const auto _ = player.PlayAsync(0, 1, true);
+        }
+    }
+    else if (!IsIndeterminate())
```

Comment:
>IsIndeterminate [](start = 14, length = 15)

I am surprised that you need to check IsIndeterminate() again. If it's intentional it needs a comment explaining how it's possible.

Reply:
I think I wrote it as an else if statement initially since there will be more state properties for progress ring to mirror progress bar. I can change it to an else statement.